app/views/admin/users/_form.html.erb

```diff
@@ -24,6 +24,26 @@
         </a>
       </li>
     `);
+
+    const applicationOnChange = () => {
+      const enabled = $('#add-application-select').val() && $(`#application-users-list input[type=\"hidden\"][name*=\"application_id\"][value=\"${$('#add-application-select').val()}\"]`).length === 0;
+
+      $('#add-application').prop('disabled', !enabled);
+    };
+
+    const applicationOnClick = (event) => {
+      event.preventDefault();
+      const nextIndex = Math.max(-1, ...Object.values($('#application-users-list input')).map(i => i.name?.match(/^application_users\[(\d+)\]/)).filter(m => m).map(m => parseInt(m[1], 10))) + 1;
```

Comment:
I was counting the number of applications before but in case they repeatedly add and delete applications it's better to take the max index + 1. Gaps seem to be fine.